src/Orleans/Streams/Internal/StreamSubscriptionHandleImpl.cs

```diff
@@ -16,11 +16,12 @@ internal class StreamSubscriptionHandleImpl<T> : StreamSubscriptionHandle<T>, IS
         private IAsyncObserver<T> observer;
         [NonSerialized]
         private StreamHandshakeToken expectedToken;
-
+        private string providerName;
```

Comment:
This will still not be backwards compatible, since it adds a new field. I would recommend making a get-only property which returns `this.streamImpl.ProviderName`